exporter/exporterhelper/internal/queuebatch/persistent_queue.go

```diff
@@ -229,6 +288,67 @@ func (pq *persistentQueue[T]) unrefClient(ctx context.Context) error {
 	return nil
 }
 
+// marshalQueueMetadata serializes the queue metadata.
+func marshalQueueMetadata(meta *queueMetadata, buf []byte) ([]byte, error) {
```

Comment:
It should always have the correct size, not sure you need to return it, this signature suggests that this is not the case. May also want to keep the size calculation within this func.